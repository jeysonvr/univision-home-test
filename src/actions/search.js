import { types } from "../types/types";

export const setKeyword = (keyword, offset) => {
    
    return (dispatch) => {
        dispatch( {
            type: types.setParameters,
            payload: {
                keyword: keyword,
                offset: offset
            }
        });
    }
}

export const setFullImage = (url, title) => {
    
    return (dispatch) => {
        dispatch( {
            type: types.setFullImage,
            payload: {
                imgURL: url,
                imgTitle: title
            }
        });
    }
}

export const setScrollSlider = (scroll) => {

    return (dispatch) => {
        dispatch( {
            type: types.setScrollSlider,
            payload: {
                scroll: scroll
            }
        });
    }

}