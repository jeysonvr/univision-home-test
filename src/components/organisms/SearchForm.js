import React from 'react';
import { NormalForm } from '../molecules/NormalForm';

export const SearchForm = ({ handleSearch }) => {

    return (
        <>
            <NormalForm
                handleSearch={ handleSearch }
            />
        </>
    )
}
