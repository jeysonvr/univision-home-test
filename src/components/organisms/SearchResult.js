import React from 'react';
import {  useDispatch ,useSelector } from 'react-redux';
import { setFullImage, setKeyword } from '../../actions/search';
import { GridItems } from '../molecules/GridItems';
import { ItemFull } from '../molecules/ItemFull';

export const SearchResult = () => {

    const dispatch = useDispatch();
    const { keyword, offset, imgURL, imgTitle } = useSelector(state => state.search);

    const handleRequiredImage = () => {
        dispatch( setKeyword(keyword, offset + 10) );
    }

    const handleCloseModal = () => {
        dispatch( setFullImage('','') );
    }

    return (
        <>
            <GridItems
                keyword={ keyword }
                offset = { offset }
                handleRequiredImage={ handleRequiredImage }
            />

            {
                (imgURL) 
                && <ItemFull 
                        url={ imgURL }
                        imgTitle={ imgTitle }
                        handleCloseModal={ handleCloseModal }
                    />
            }
        </>
    )
}
