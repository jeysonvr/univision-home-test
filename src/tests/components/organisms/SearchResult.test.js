import React from 'react';
import { mount } from "enzyme";
import { Provider } from "react-redux";

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';

import { SearchResult } from '../../../components/organisms/SearchResult';
import { setFullImage } from '../../../actions/search';

jest.mock('../../../actions/search', () => ({
    setFullImage: jest.fn()
}))

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
    search: {
        keyword: 'test', 
        offset: 0, 
        imgURL: 'a', 
        imgTitle: 'b'
    }
};
let store = mockStore(initialState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={ store }>
        <SearchResult />
    </Provider>
);

describe('Test in <SearchForm />', () => {

    beforeEach( () => {
        store = mockStore(initialState);
        jest.clearAllMocks();
    })

    test('should show component correctly', () => {
        expect( wrapper ).toMatchSnapshot(); 
    })

    test('should trigger action handleCloseModal', () => {
        wrapper.find('.button__close').prop('onClick')();
        expect( setFullImage ).toHaveBeenCalled();
    })

})