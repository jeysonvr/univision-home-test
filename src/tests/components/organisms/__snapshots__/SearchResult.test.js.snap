// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SearchForm /> should show component correctly 1`] = `
Array [
  Array [
    <p
      className="card"
    >
      Loading
    </p>,
    <div
      className="alert alert-primary m-5"
      role="alert"
    >
      Search something ...
    </div>,
  ],
  Array [
    <div
      className="modal animate__animated animate__fadeIn"
      style={
        Object {
          "display": "block",
        }
      }
      tabIndex="-1"
    >
      <div
        className="modal-dialog modal-dialog-centered"
      >
        <div
          className="modal-content text-center"
        >
          <div
            className="modal-header"
          >
            <h5
              className="modal-title"
            >
              b
            </h5>
            <button
              className="btn button__close"
              onClick={[Function]}
            >
              <i
                className="fas fa-times-circle fa-2x"
              />
            </button>
          </div>
          <div
            className="modal-body card"
          >
            <img
              alt="b"
              className="img-fluid"
              src="a"
            />
          </div>
        </div>
      </div>
    </div>,
    <div
      className="modal-backdrop fade show"
    />,
  ],
]
`;
